src/ControlSystem/ExpirationTimes.hpp

```diff
@@ -0,0 +1,140 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <unordered_map>
+
+#include "ControlSystem/Tags.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "Domain/Creators/DomainCreator.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+template <size_t DerivOrder>
+class Controller;
+class TimescaleTuner;
+/// \cond
+
+namespace control_system {
+/// \cond
+template <size_t DerivOrder>
+DataVector calculate_measurement_timescales(
+    const ::Controller<DerivOrder>& controller, const ::TimescaleTuner& tuner,
+    const int measurements_per_update);
```

Comment:
Sorry if this was here before and I missed it, but if you're going to do it this way you should explicitly instantiate `calculate_measurement_timescales` so that this file doesn't depend on the details of implementations of functions elsewhere causing implicit instantiations.  Better would be to break the include loop.